Mathlib/CategoryTheory/Distributive/Monoidal.lean

```diff
@@ -0,0 +1,321 @@
+            apply PreservesColimitPair.of_iso_coprod_comparison (tensorRight X) _ _
+        apply preservesColimit_of_iso_diagram (tensorRight X) (diagramIsoPair K).symm
+    }
+
+variable [IsMonoidalRightDistrib C]
+
+/-- The forward direction of the right distributivity isomorphism is equal to the cogap morphism
+`coprod.desc (coprod.inl ▷ _) (coprod.inr ▷ _) : (Y ⊗ X) ⨿ (Z ⊗ X) ⟶ (Y ⨿ Z) ⊗ X`. -/
+@[simp]
+lemma rightDistrib_hom {X Y Z : C} :
+  (∂R X Y Z).hom = coprod.desc (coprod.inl ▷ _) (coprod.inr ▷ _) := by rfl
+
+lemma coprod_inl_rightDistrib_hom {X Y Z : C} :
+    coprod.inl ≫ (∂R X Y Z).hom = (coprod.inl ▷ X) := by
+  rw [rightDistrib_hom, coprod.inl_desc]
+
+lemma coprod_inr_rightDistrib_hom {X Y Z : C} :
+    coprod.inr ≫ (∂R X Y Z).hom = (coprod.inr ▷ X) := by
+  rw [rightDistrib_hom, coprod.inr_desc]
+
+/-- The composite of `(coprod.inl ▷ X) : Y ⊗ X ⟶ (Y ⨿ Z) ⊗ X` and
+`(∂R X Y Z).inv :  (Y ⨿ Z) ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)` is equal to the left coprojection
+`coprod.inl : Y ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)`. -/
+@[reassoc (attr := simp)]
+lemma whisker_inl_inv_rightDistrib {X Y Z : C} :
+    (coprod.inl ▷ X) ≫ (∂R X Y Z).inv = coprod.inl := by
+  apply (cancel_iso_hom_right _ _ (∂R X Y Z)).mp
+  rw [assoc, Iso.inv_hom_id, comp_id, coprod_inl_rightDistrib_hom]
+
+/-- The composite of `(coprod.inr ▷ X) : Z ⊗ X ⟶ (Y ⨿ Z) ⊗ X` and
+`(∂R X Y Z).inv :  (Y ⨿ Z) ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)` is equal to the right coprojection
+`coprod.inr : Z ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)`. -/
+@[reassoc (attr := simp)]
+lemma whisker_inr_inv_rightDistrib {X Y Z : C} :
+    (coprod.inr ▷ X) ≫ (∂R X Y Z).inv = coprod.inr := by
+  apply (cancel_iso_hom_right _ _ (∂R X Y Z)).mp
+  rw [assoc, Iso.inv_hom_id, comp_id, coprod_inr_rightDistrib_hom]
+
+end IsMonoidalRightDistrib
+
+/-- In a symmetric monoidal category, the left distributivity is equal to
+the right distributivity up to braiding isomorphisms. -/
+@[simp]
+lemma SymmetricCategory.leftDistrib_braiding [SymmetricCategory C] {X Y Z : C} :
+    (coprodComparison (tensorLeft X) Y Z) ≫ (β_ X (Y ⨿ Z)).hom =
+    (coprod.map (β_ X Y).hom (β_ X Z).hom) ≫ (coprodComparison (tensorRight X) Y Z) := by
+  simp [coprodComparison]
+
+
+/-- In a symmetric monoidal category, the right distributivity is equal to
+the left distributivity up to braiding isomorphisms. -/
+@[simp]
+lemma SymmetricCategory.rightDistrib_braiding [SymmetricCategory C] {X Y Z : C} :
+    (coprodComparison (tensorRight X) Y Z) ≫ (β_ (Y ⨿ Z) X).hom =
+    (coprod.map (β_ Y X).hom (β_ Z X).hom) ≫ (coprodComparison (tensorLeft X) Y Z) := by
+  simp [coprodComparison]
+
+/-- A left distributive symmetric monoidal category is right distributive. -/
+instance SymmetricCategory.isMonoidalRightDistrib_of_isMonoidalLeftDistrib
+    [SymmetricCategory C] [IsMonoidalLeftDistrib C] : IsMonoidalRightDistrib C where
+  preservesBinaryCoproducts_tensorRight X :=
+    preservesColimitsOfShape_of_natIso (SymmetricCategory.tensorLeftIsoTensorRight X)
```

Comment:
The definition `SymmetricCategory.tensorLeftIsoTensorRight` should be generalized to `BraidedCategory.tensorLeftIsoTensorRight`:
```lean
/-- In a braided monoidal category, the functors `tensorLeft X` and
`tensorRight X` are isomorphic. -/
def BraidedCategory.tensorLeftIsoTensorRight {C : Type u} [Category.{v} C]
    [MonoidalCategory.{v} C] [BraidedCategory C] (X : C) :
    tensorLeft X ≅ tensorRight X where
  hom := { app Y := (β_ X Y).hom }
  inv := { app Y := (β_ X Y).inv }
```